pkg/v2/resource/chart/create_test.go

```diff
@@ -0,0 +1,117 @@
+package chart
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/spf13/afero"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
+)
+
+func Test_Resource_Chart_newCreate(t *testing.T) {
+	obj := v1alpha1.ClusterGuestConfig{
+		DNSZone: "5xchu.aws.giantswarm.io",
+		ID:      "5xchu",
+		Owner:   "giantswarm",
+	}
+
+	testCases := []struct {
+		currentState  *ResourceState
+		desiredState  *ResourceState
+		expectedState *ResourceState
+		description   string
+	}{
+		{
+			description:   "empty current and desired, expected empty",
+			currentState:  &ResourceState{},
+			desiredState:  &ResourceState{},
+			expectedState: &ResourceState{},
+		},
+		{
+			description: "non-empty current, empty desired, expected empty",
+			currentState: &ResourceState{
+				ChartName: "current",
+			},
+			desiredState:  &ResourceState{},
+			expectedState: &ResourceState{},
+		},
+
+		{
+			description:  "empty current, non-empty desired, expected desired",
+			currentState: &ResourceState{},
+			desiredState: &ResourceState{
+				ChartName: "desired",
+			},
+			expectedState: &ResourceState{
+				ChartName: "desired",
+			},
+		},
+		{
+			description: "equal non-empty current and desired, expected empty",
+			currentState: &ResourceState{
+				ChartName: "desired",
+			},
+			desiredState: &ResourceState{
+				ChartName: "desired",
+			},
+			expectedState: &ResourceState{},
+		},
+		{
+			description: "different non-empty current and desired, expected empty",
+			currentState: &ResourceState{
+				ChartName: "current",
+			},
+			desiredState: &ResourceState{
+				ChartName: "desired",
+			},
+			expectedState: &ResourceState{},
+		},
```

Comment:
Why in this case expectedState is empty if the desired has a chart that is not in the current state?

Reply:
The method added here is for creating the resource instance, AIUI if the desired state is different from the current state the mismatch should be reconciled as part of the update logic.